automation/tests-storage.sh

```diff
@@ -12,28 +24,50 @@ create_loop_devices() {
 }
 
 setup_storage() {
-    python3 tests/storage/userstorage.py setup
+    # Configure lvm to ignore udev events, otherwise some lvm tests hang.
+    mkdir -p /etc/lvm
+    cp docker/lvmlocal.conf /etc/lvm/
+
+    # Make sure we have enough loop device nodes. Using 16 devices since with 8
+    # devices we have random mount failures.
+    create_loop_devices 16
+
+    # Build vdsm.
+    ./autogen.sh --system
+    make
+
+    # Setup user storage during the tests.
+    make storage
 }
 
 teardown_storage() {
-    python3 tests/storage/userstorage.py teardown \
+    make clean-storage \
         || echo "WARNING: Ingoring error while tearing down user storage"
 }
 
-# Configure lvm to ignore udev events, otherwise some lvm tests hang.
-mkdir -p /etc/lvm
-cp docker/lvmlocal.conf /etc/lvm/
-
-# Make sure we have enough loop device nodes. Using 16 devices since with 8
-# devices we have random mount failures.
-create_loop_devices 16
+# Process user argument
+user=$1
+if [ -z "$user" ]; then
+    echo "ERROR: user required"
+    print_help
+    exit 1
+fi
 
-# Build vdsm.
-./autogen.sh --system
-make
+echo "Running tests as user $user"
 
-# Setup user stoage during the tests.
-trap teardown_storage EXIT
 setup_storage
+if [ "$user" != "root" ]; then
+    # Change ownership of current and storage folders
+    # to allow non-privileged access.
+    chown -R vdsm:kvm ./ /var/tmp/vdsm*
+fi
+# Teardown storage before exit.
+trap teardown_storage EXIT
 
-make tests-storage
+if [ "$user" = "root" ]; then
+    # Run only tests marked as root
+    make tests-storage-root
+else
+    # Run tests not marked as root as $user
+    su -c "make tests-storage-user" -- ${user}
```

Comment:
Maybe:

    sudo -u $user make tests-storage-user

Reply:
It does not preserve the $PATH, to make it work it would have to be:
```
sudo -u $user env "PATH=$PATH" make tests-storage-user
```
I don't mind which option goes here.

Reply:
It also loses all other env variables, which ends up making the test to ignore the `broken_on_ci` fixtures and fail: https://github.com/aesteve-rh/vdsm/runs/7355846727

I think it is better if we stick with the current strategy.

Reply:
Good points, we want to keep the environment. Would be nice to add a comment why we use su instead of sudo.